LIPS/lip-25.md

```diff
@@ -0,0 +1,1983 @@
+/// @return targetValidatorsCount relative target active validators limit for operator
+/// @return stuckValidatorsCount number of validators with an expired request to exit time
+/// @return refundedValidatorsCount number of validators that can't be withdrawn, but deposit
+///     costs were compensated to the Lido by the node operator
+/// @return stuckPenaltyEndTimestamp time when the penalty for stuck validators stops applying
+///     to node operator rewards
+/// @return totalExitedValidators total number of validators in the EXITED state
+///     on the Consensus Layer. This value can't decrease in normal conditions
+/// @return totalDepositedValidators total number of validators deposited via the official
+///     Deposit Contract. This value is a cumulative counter: even when the validator goes into
+///     EXITED state this counter is not decreasing
+/// @return depositableValidatorsCount number of validators in the set available for deposit
+function getNodeOperatorSummary(uint256 _nodeOperatorId) external view returns (
+    uint256 targetLimitMode,
+    uint256 targetValidatorsCount,
+    uint256 stuckValidatorsCount,
+    uint256 refundedValidatorsCount,
+    uint256 stuckPenaltyEndTimestamp,
+    uint256 totalExitedValidators,
+    uint256 totalDepositedValidators,
+    uint256 depositableValidatorsCount
+);
+
+/// @notice Updates the number of the validators in the EXITED state for node operator with given id
+/// @param _nodeOperatorIds bytes packed array of the node operators id
+/// @param _exitedValidatorsCounts bytes packed array of the new number of EXITED validators for the node operators
+function updateExitedValidatorsCount(
+    bytes calldata _nodeOperatorIds,
+    bytes calldata _exitedValidatorsCounts
+) external;
+
+/// @notice Updates the limit of the validators that can be used for deposit
+/// @param _nodeOperatorId Id of the node operator
+/// @param _targetLimitMode taret limit mode
+/// @param _targetLimit Target limit of the node operator
+function updateTargetValidatorsLimits(
+    uint256 _nodeOperatorId,
+    uint256 _targetLimitMode,
+    uint256 _targetLimit
+) external;
+```
+
+#### 1.3. New events
+
+Two new events should be added to the `IStakingModule` interface.
+
+```solidity
+/// @dev Event to be emitted when a signing key is added to the StakingModule
+event SigningKeyAdded(uint256 indexed nodeOperatorId, bytes pubkey);
+
+/// @dev Event to be emitted when a signing key is removed from the StakingModule
+event SigningKeyRemoved(uint256 indexed nodeOperatorId, bytes pubkey);
+```
+
+### 2. `StakingRouter` contract changes
+
+All the code of this contract assumes the Solidity v0.8.9 syntax.
+
+#### 2.1. Module's share consideration for validator exits
+
+Staking Router now should consider module's share when prioritizing validators for exit. Support of this improvement in the `StakingRouter` contract requires several changes:
```

Comment:
It's more correct to say that VEBO is responsible for prioritization, but target share and exit threshold params are stored in SR

Reply:
Yes, agree. I've updated this section with a much more detailed explanation. Your note has also been taken into account.